### PR TITLE
feat(api): add gate to allow manual trigger of a job

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-cds#19711.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19711.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19713.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19713.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19714.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19714.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19716.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19716.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19717.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19717.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19718.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19719.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19720.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19720.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19767.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19767.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19767.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

File: engine/api/v2_workflow_run.go [hunk 1]
```diff
@@ -791,93 +794,278 @@ func (api *API) putWorkflowRunV2Handler() ([]service.RbacChecker, service.Handle
...
+			if _, ok := interpolatedInput.(string); !ok {
+				return sdk.NewErrorFrom(sdk.ErrInvalidData, "job %s: gate statement does not return a string. Got %v", jobToRun.JobID, interpolatedInput)
+			}
+
+			booleanResult, err := strconv.ParseBool(interpolatedInput.(string))
+			if err != nil {
+				return sdk.NewErrorFrom(sdk.ErrInvalidData, "job %s: gate statement does not return a boolean. Got %s", jobToRun.JobID, interpolatedInput)
+			}
+			if !booleanResult {
+				return sdk.NewErrorFrom(sdk.ErrForbidden, "gate condition are not satisfied")
```
Comment (fsamin) [inline engine/api/v2_workflow_run.go:1028]:
```suggestion
				return sdk.NewErrorFrom(sdk.ErrForbidden, "gate conditions are not satisfied")
```

**Review by yesnault — APPROVED** [review]

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19797.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19797.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19797.1 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19807.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19807.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by fsamin — APPROVED** [review]

**Pull request merged**